_posts/2020-01-06-using-strimzi-with-amazon-nlb-loadbalancers.md

```diff
@@ -0,0 +1,348 @@
+The only different is that this time instead of the load balancer IP address we have to use the CIDR range from which our clients will connect.
+In my case, the client will connect from `88.208.0.0/16`.
+
+[![Add the security group rules]({{ "/assets/2020-01-06-nlb-security-group-rules.png" }})](/assets/2020-01-06-nlb-security-group-rules.png)
+
+Once you have all the rules prepared, don't forget to press the _Save_ button to save the changes.
+Once you save the rules, we should have the Network load balancer prepared.
+But we still need to configure the Kafka broker to use the load balancer as the advertised address.
+
+_Note: the Network load balancer can be of course configures also using the different AWS command line tools._
+
+### Configuring the advertised addresses
+
+The advertised address is a configuration in the Kafka brokers which is used to tell the clients how to connect to the broker.
+Since we configured the Kafka cluster with node ports, it will by default use the address of the Kubernetes node and the node port of the service as the advertised address.
+We need to reconfigure the advertised address to route the traffic through the load balancer.
+To do so, we will need to grab the load balancer address from the AWS console.
+In my case, the address is `my-cluster-kafka-36203843994a2519.elb.us-east-1.amazonaws.com`.
+To configure the advertised address, we will need to edit the Kafka custom resource and set the [advertised listener overrides](https://strimzi.io/docs/latest/full.html#con-kafka-broker-external-listeners-addresses-deployment-configuration-kafka) to the load balancer address and to the listener ports (in our case `9093`, `9094` and `9095`):
+
+```yaml
+# ...
+listeners:
+  # ...
+  external:
+    type: nodeport
+    tls: false
+    overrides:
+      brokers:
+      - broker: 0
+        advertisedHost: my-cluster-kafka-36203843994a2519.elb.us-east-1.amazonaws.com
+        advertisedPort: 19092
+      - broker: 1
+        advertisedHost: my-cluster-kafka-36203843994a2519.elb.us-east-1.amazonaws.com
+        advertisedPort: 29092
+      - broker: 2
+        advertisedHost: my-cluster-kafka-36203843994a2519.elb.us-east-1.amazonaws.com
+        advertisedPort: 39092
+# ...
+```
+
+Once we change the configuration, a rolling update will be needed to reconfigure the brokers.
+
+### Connecting with clients
+
+Once the rolling update is finished, we should be ready.
+We can try to connect with our client:
+
+```
+$ bin/kafka-console-consumer.sh --bootstrap-server my-cluster-kafka-36203843994a2519.elb.us-east-1.amazonaws.com:9092 --topic my-topic --from-beginning
+Hello
+World
+```
+
+## Are there any real advantages?
+
+When it comes to Strimzi and Kafka, the main advantages of the Network load balancers over the Classic LBs would be probably the price and the possibility to use a single load balancer instance for many listeners.
```

Comment:
I would use "Classic load balancers" instead of "Classic LBs" ... or just "...over the Classic ones" ?